ykrt/src/compile/jitc_yk/jit_ir.rs

```diff
@@ -211,13 +211,14 @@ impl Module {
     }
 
     /// Push an instruction to the end of the [Module].
-    ///
-    /// # Panics
-    ///
-    /// If `instr` would overflow the index type.
-    pub(crate) fn push(&mut self, inst: Inst) {
-        assert!(InstIdx::new(self.insts.len()).is_ok());
-        self.insts.push(inst);
+    pub(crate) fn push(&mut self, inst: Inst) -> Result<InstIdx, CompilationError> {
+        match InstIdx::new(self.insts.len()) {
```

Comment:
Can this be:
```
InstIdx::new(self.insts.len()).and_then(|i| {
    self.insts.push(inst);
    Ok(i)
})
```

Reply:
Yeah, I experimented with that, but I wasn't sure it was clearer with the mutation? I'm easy though.

Reply:
Neither way is fantastic. At least this is shorter? Won't die on that hill.

Reply:
I'm inclined to leave it as-is TBH.